### ISSUE TITLE
Opt-in for error logging (talkback service)

### ISSUE THREAD

**Issue by ktomala** [golemfactory/clay#2298]
Since we have currently set up service for talkback in Golem, some changes must be done in logging configuration of golemapp to start using this functionality.

First of all we should create a configuration parameter (possibly in config file handler) for talkback opt-in. This parameter must be allowed to change via RPC from golem-electron app. When this parameter changes logging configuration should be updated and reload should occur. 

Since LOGGING variable is dictionary we can update specific keys adding one handler (possibly requires trigger on parameter change - can be implemented in this method https://github.com/golemfactory/golem/blob/73c6c4fac0ba66ab34390eecd771dec61ef6de99/golem/client.py#L852):

```
    'handlers': {
        ...
        'sentry': {
            'level': 'ERROR',
            'class': 'raven.handlers.logging.SentryHandler',
            'dsn': 'https://<key>:<secret>@talkback.golem.network/<project>',
            },
        ...
    }
```

and enabling it in loggers configuration:

```
    'loggers': {
        '': {
            'handlers': ['console', 'sentry'],
            ...
         }
    }
```

Additional dependency must be installed for this to work:

```pip install raven```

and possibly also put into setup.py and/or requirements.txt.

In case we want to exclude some loggers (e.g. for privacy reasons) we can use `exclude` option in handler configuration like this:

```
from raven.conf import setup_logging

setup_logging(handler, exclude=("logger1", "logger2", ...))
```

Logging configuration reloading can be done with e.g. adding default parameter `reload=False` to this method: https://github.com/golemfactory/golem/blob/73c6c4fac0ba66ab34390eecd771dec61ef6de99/golem/core/common.py#L166 and adding if's where necessary.

**Comment by ktomala** [comment]
If we are concerned about users privacy we can also create a custom logging formatter that will scrub unwanted data from logs sent to our talkback service. Otherwise the data scrubbing is also implemented server-side.

**Comment by maaktweluit** [comment]
Whoops, commented on electron, will repeat it here since its more applicable here:

We could use the `--nomonitor` `golemapp` argument for this, currently by default data is send to the monitor.

**Comment by ktomala** [comment]
As I have stated in golem-electron: `This is something else, I believe we should have a distinctive option. And since this is opt-in (not opt-out) we would prefer rather --enable-talkback or something like this.`

**Comment by ktomala** [comment]
Will be resolved by merging https://github.com/golemfactory/golem/pull/2375

**Comment by ktomala** [comment]
This issue was completed some time ago.
